### PR TITLE
MOSIP-21376 Update existing api checkAIDStatus

### PR DESCRIPTION
**Pull request by kameshsr**

### CODE REVIEW

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentServiceImpl.java [hunk 1]
```diff
@@ -1354,4 +1369,94 @@ public String checkAidStatus(String aid) throws ResidentServiceCheckedException
...
+			eventStatus = EventStatus.IN_PROGRESS.name();
+		}
+		return eventStatus;
+	}
+
+	private String getResidentTransactionEntityValue(Optional<ResidentTransactionEntity> residentTransactionEntity, String key) {
+		String value = null;
+		switch (key) {
+
+		case "purpose":
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentServiceImpl.java]: Please create constats for these strings

File: resident/resident-service/src/main/java/io/mosip/resident/util/EventEnum.java [hunk 2]
```diff
@@ -601,7 +601,11 @@ public enum EventEnum {
 	GET_TEMPLATES_EXCEPTION("RES-SER-247", RegistrationConstants.SYSTEM, "get templates failure",
 			"get templates by langCode and templateTypeCode is failed", "RES-SER", "Residence service", "NO_ID",
 			"NO_ID_TYPE", RegistrationConstants.APPLICATIONID, RegistrationConstants.APPLICATIONNAME),
+	INVALID_LANGUAGE_CODE("RES-SER-500", RegistrationConstants.SYSTEM, "Invalid language code",
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/util/EventEnum.java:604]: please document the error code and message to the API story

File: resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java [hunk 3]
```diff
@@ -13,47 +14,156 @@
 @Component
  public class TemplateUtil {
 
-    public Map<String, String> getAckTemplateVariablesForCredentialShare(String eventId){
-        return Collections.emptyMap();
+    public Map<String, String> getAckTemplateVariablesForCredentialShare(String eventId) {
+        Map<String, String> templateVariables = new HashMap<>();
+        templateVariables.put("eventId", eventId);
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java]: Please create constants for the string

File: resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java [hunk 4]
```diff
@@ -13,47 +14,156 @@
...
+        templateVariables.put("eventType", "credential-share");
+        templateVariables.put("purpose", "credential-share");
+        templateVariables.put("eventStatus", "success");
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java]: The values are to be calculated from the request/session. For all the features..

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentServiceImpl.java (hunk 1 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentServiceImpl.java]: yes done.

File: resident/resident-service/src/main/java/io/mosip/resident/util/EventEnum.java (hunk 2 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/util/EventEnum.java:604]: yes done.

File: resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java (hunk 3 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java]: yes done.

File: resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java (hunk 4 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/util/TemplateUtil.java]: Yes Fixed for these type, i am deciding values in service class only taking variable name from template variable method in TemplateUtil class.

**Review by loganathan-sekaran — APPROVED** [review]

**Pull request merged**